src/main/preference.js

```diff
@@ -57,6 +65,28 @@ class Preference {
       })
     })
   }
+
+  update () {
```

Comment:
Another question, I still think you should delete the outdated user setting and add the new use setting to `preferences.md` in `userDataPath `.

eg,

this old user preferences is:
```json
color: 'red',
theme: 'dark'
```

and the new user preferences is:

```json
darkColor: '#ffffff',
lightColor: '#000000',
theme: 'light'
```

The `color` in the setting is out of date, so we should delete it, and add `darkColor` and `lightColor` to `preferences.md`.

is it right ?

Reply:
I didn't know that we have outdated entries. Ok... let's say we create a array with a pair of `oldKeyName` and a function like this:

```
function (oldKey, oldValue) {
  // update key
  return newKey
}
```

Then we can replace old keys and/or convert them.

Reply:
My fault, my before comments are not clear.